packages/halo/src/react/index.js

```diff
@@ -0,0 +1,73 @@
+import * as glamor from 'glamor'
+import polyfillFocusWithin from 'focus-within'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import css, { BASE_CLASSNAME } from '../css'
+import * as vars from '../vars'
+
+import withDefaultTheme from './with-default-theme'
+
+polyfillFocusWithin(document)
+
+const styles = {
+  halo: props => {
+    const theme = `${BASE_CLASSNAME}--theme-${props.themeName}`
+    const appearance = `${BASE_CLASSNAME}--appearance-${props.appearance}`
+    const gapSize = `${BASE_CLASSNAME}--gap-size-${props.gapSize}`
+    const shape = `${BASE_CLASSNAME}--shape-${props.shape}`
+
+    return glamor.css(
```

Comment:
there seems to be some hurdles to getting the write css for the components. also i think this is generating a classname for every combination of possible variations. i assume there is a reason i was just curious if there is a simplier way